sourcefiles/elementrando.py

```diff
@@ -0,0 +1,550 @@
+from __future__ import annotations
+
+from typing import Dict
+import re
+
+import ctrom
+from ctenums import Element as El, TechID as T, LocID as L, CharID as C
+from techdb import TechDB
+import ctstrings
+
+import randoconfig as cfg
+import randosettings as rset
+
+def write_config(settings: rset.Settings, config: cfg.RandoConfig, rand):
+    if rset.GameFlags.ELEMENT_RANDO in settings.gameflags:
+        elems = [El.LIGHTNING, El.SHADOW, El.ICE, El.FIRE]
+        doubled_elem = rand.choice(elems)
+
+        roboelems = list(set(elems) - {doubled_elem,})
+        elems = elems + [doubled_elem]
+            
+        # These are just keyed by order:
+        # crono marle lucca (skip robo) frog (skip ayla) magus
+        # laser spin, area bomb, shock for robo
+        rand.shuffle(elems) 
+        rand.shuffle(roboelems)
+
+        for i, elem in enumerate(elems):
+            real_i = i
+            if i == 3:
+                real_i = 4
+            elif i == 4:
+                real_i = 6
+            statblock = config.pcstats.pc_stat_dict[C(real_i)].stat_block
+            statblock.innate_element_resisted = elem
+
+        config.elems = elems + roboelems
+        config.tech_db = shuffle_techdb(config.tech_db, elems, roboelems)
+
+def update_ctrom(ct_rom: ctrom.CTRom, config: cfg.RandoConfig):
+    update_scripts(ct_rom, config)
```

Comment:
I added this function that doesn't do much in anticipation of adding another function or some lines to this one for the menu placard stuff.